hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/node/NodeDecommissionManager.java

```diff
@@ -375,4 +376,20 @@ private NodeStatus getNodeStatus(DatanodeDetails dn)
     return nodeManager.getNodeStatus(dn);
   }
 
+  /**
```

Comment:
We should probably add a test for this - ie ensure that nodes are added to the decommission workflow when the onBecomeLeader() event is fired.